autoPyTorch/api/base_task.py

```diff
@@ -196,14 +196,6 @@ def __init__(
                 raise ValueError("Expected search space updates to be of instance"
                                  " HyperparameterSearchSpaceUpdates got {}".format(type(self.search_space_updates)))
 
-    @abstractmethod
-    def _get_required_dataset_properties(self, dataset: BaseDataset) -> Dict[str, Any]:
```

Comment:
it would be nice if we have `DatasetInfo` object to refer to the information we should include.
Btw, which name would you prefer, either `dataset_properties` or `required_dataset_info`?

Reply:
I know it would be useful to have these objects but I don't see them as a priority. As discussed in the meeting on Wednesday, we should aim to get a new release by next week. Therefore, for the old code that's already been put in, I feel we can start refactoring these into `AttrDict` objects once we are done with the reproducibility results. We can make sure any new code we put, is made with such `AttrDict` that makes the code easier to understand but changing old code should not be our priority right now. I suggest you make an issue for them or as we already have this issue  #140, you can add any problem we have as a comment there.

Reply:
Sorry, I forgot responding you for such a long time!!!
I did not say you have to put AttrDict, but can you put NamedTuple, right?
Anyways, the contents are not clear to me especially the exact key names.

Reply:
I may have mentioned only AttrDict but my comment was for both. Although I would like to clarify what exactly is the issue, is it about the dataset properties that are passed into the fit dictionary? Or the data set info needed to instantiate pipeline. In terms of knowing the exact key names, this is something the user or even we do not have to directly interact with or change the key names. The issue I have with making a AttrDict or namedTuple, is we can not really define what would be needed in a pipeline. For different pipelines, there would be different dataset properties based on the components that are active in it. I can enhance the documentation to explain these terms better, but in its current form, we need it to be a mutable object.